kpops/manifests/argo.py

```diff
@@ -0,0 +1,50 @@
+import enum
+from typing import Any
+
+from typing_extensions import override
+
+
+class ArgoEnricher:
+    @property
+    def key(self) -> str:
+        return NotImplemented
+
+    @property
+    def value(self) -> str:
+        return NotImplemented
+
+    def enrich(self, helm_values: dict[str, Any]) -> dict[str, Any]:
+        annotations = helm_values.setdefault("annotations", {})
+        annotations[self.key] = self.value
+        return helm_values
+
+
+class ArgoHook(ArgoEnricher, str, enum.Enum):
+    POST_DELETE = "PostDelete"
+
+    @property
+    @override
+    def key(self) -> str:
+        return "argocd.argoproj.io/hook"
+
+    @property
+    @override
+    def value(self) -> str:
+        return self.POST_DELETE._value_
```

Comment:
```suggestion
        return self.value
```

Reply:
The reason I did this is because `self.value` points to the value field and causes 
```
E   RecursionError: maximum recursion depth exceeded
!!! Recursion detected (same locals & position)
```

Reply:
oh ok, please avoid overriding the builtin `value` attribute of Enum